homework/index.js

```diff
@@ -16,31 +37,139 @@
+    fetchJSON(url, (err, data) => {
+      if (err) {
+        renderError(err);
+        return;
       }
+      data.forEach(element => {
+        const ContributorEl = createAndAppend('a', infoRight, {
+          class: 'Contributor_el',
+          href: element.html_url,
+          target: '_blank',
+        });
+        const imgDiv = createAndAppend('div', ContributorEl, { class: 'Contributor_img_div' });
+        createAndAppend('img', imgDiv, {
+          class: 'Contributor_img',
+          src: element.avatar_url,
+          alt: 'Contributor photo',
+          width: '50px',
+        });
+        const dataDiv = createAndAppend('div', ContributorEl, { class: 'Contributor_data_div' });
+        createAndAppend('div', dataDiv, {
+          class: 'Contributor_name',
+          text: element.login,
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'Contributor_contributions',
+          text: element.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr_contributor',
+        });
+      });
+    });
+  }
+
+  function viewRepInfo() {
+    fetchJSON(HYF_REPOS_URL, (err, data) => {
+      if (err) {
+        renderError(err);
+        return;
+      }
+      const repoName = document.getElementById('repositories_list').value;
+      const wantedRepo = data.filter(element => element.name === repoName)[0];
+      const infoContainer = document.getElementById('info_container');
+      infoContainer.innerHTML = '';
+      const infoLeft = createAndAppend('div', infoContainer, {
+        class: 'info_left',
+        text: 'Repository information:',
+      });
+      const nameAndLink = createAndAppend('p', infoLeft, {
+        class: 'name_ink',
+        text: 'Name:',
+      });
+      createAndAppend('a', nameAndLink, {
+        class: 'rep_link',
+        text: wantedRepo.name,
+        href: wantedRepo.html_url,
+        target: '_blank',
+      });
+      const Description = createAndAppend('p', infoLeft, {
+        class: 'p_infoLeft',
+        text: 'Description:',
+      });
+      createAndAppend('span', Description, {
+        class: 'span_infoLeft',
+        text: wantedRepo.description,
+      });
+      const forks = createAndAppend('p', infoLeft, {
+        class: 'p_infoLeft',
+        text: 'Forks:',
+      });
+      createAndAppend('span', forks, {
+        class: 'span_infoLeft',
+        text: wantedRepo.forks,
+      });
+      const LastUpdate = createAndAppend('p', infoLeft, {
+        class: 'p_infoLeft',
+        text: 'Last update:',
+      });
+      createAndAppend('span', LastUpdate, {
+        class: 'span_infoLeft',
+        text: new Date(wantedRepo.updated_at).toLocaleString(),
+      });
+      const contributorsUrl = createAndAppend('span', infoLeft, {
+        text: wantedRepo.contributors_url,
+        id: 'contributors_url',
+        class: 'hidden',
+      });
+      contributorsUrl.addEventListener('load', Contributions(contributorsUrl));
```

Comment:
The purpose of this hidden element and the event listener is unclear to me. You can simply replace lines 129-134 with

```js
Contributions(wantedRepo.contributors_url);
```

and change your `Contributions` function as follows:

```js
function Contributions(url) {
  const infoContainer = document.getElementById('info_container');
  const infoRight = createAndAppend('div', infoContainer, {
  ...
}
```

Function names should be in camelCase (except for constructor function) and include a verb that indicates the action the function perform. Your `Contributions` function could for instance be named `fethcAndRenderContributions`.

Reply:
I deleted that hidden element . and the function is adjusted.

Reply:
I should use verbs as names for function. while I was writing this homework ,I forgot all the standers of making names.
I have to be careful with that.